kubernetes/config/kube_config.py

```diff
@@ -22,151 +22,243 @@
 from kubernetes.client import configuration
 from oauth2client.client import GoogleCredentials
 
-_temp_files = []
+from .incluster_config import ConfigException
+
+_temp_files = {}
 
 
 def _cleanup_temp_files():
-    for f in _temp_files:
-        os.remove(f)
+    for temp_file in _temp_files.values():
+        os.remove(temp_file)
```

Comment:
_temp_files was array before and dict now, so I cannot do it without values() (it will give me the keys in case of dict).